include/SDL3/SDL_video.h

```diff
@@ -96,6 +96,40 @@ typedef struct SDL_DisplayMode
     void *driverdata;           /**< driver-specific data, initialize to 0 */
 } SDL_DisplayMode;
 
+/**
+ * A structure containing advanced HDR color and luminance data.
+ *
+ * \since This struct is available since SDL 3.0.0
+ *
+ * \sa SDL_HDRProperties
+ */
+typedef struct SDL_HDRColorData
+{
+    SDL_FPoint red_primary;         /**< the red primary x/y coordinates */
+    SDL_FPoint green_primary;       /**< the green primary x/y coordinates */
+    SDL_FPoint blue_primary;        /**< the blue primary x/y coordinates */
+    SDL_FPoint white_point;         /**< the white point x/y coordinates */
+    float min_luminance;            /**< the minimum luminance level in nits */
+    float max_luminance;            /**< the maximum luminance level in hits */
+    float max_full_frame_luminance; /**< the maximum full-frame luminance in nits */
+} SDL_HDRColorData;
+
+/**
+ * The structure containing the output color and luminance metadata for a display/window.
+ *
+ * \since This struct is available since SDL 3.0.0
+ *
+ * \sa SDL_GetDisplayHDRProperties
+ * \sa SDL_GetWindowHDRProperties
+ */
+typedef struct SDL_HDRProperties
+{
+    SDL_bool has_color_data;     /**< SDL_TRUE if the color_data member contains valid color point and luminance data */
+    SDL_HDRColorData color_data; /**< the advanced color and luminance properties for the output; only contains valid data if has_color_data is SDL_TRUE */
+    float SDR_white_level;       /**< the SDR white level */
+    float HDR_headroom;          /**< the additional dynamic range that can be displayed in terms of the SDR white level (greater than 1.0 if the output is EDR/HDR capable */
```

Comment:
What's the difference between the white level and headroom? The headroom works with any encoding and without knowing min/max absolute levels, but the HDR metadata luminances require knowing either the absolute level of the white level or min/max absolute levels. The white level gives the headroom if one assumes black levels are always the same and if one knows the min/max absolute levels.

IOW, both require min/max luminance levels to work properly, but once they are available, they are equal (ignoring the black level issue).

So why have both values? Why not just

```c
typedef struct SDL_HDRProperties
{
  SDL_bool has_color_data;
  SDL_HDRColorData color_data;
  float min_luminance;
  float max_luminance;
  float reference_white_level;
}
```

For the HDRColorData you're also not using headroom anyway...

Reply:
Apple platforms don't actually provide the color/white points or min/max luminance data for the window surface or display (if there is a way to get it without parsing the EDID, it's not obvious), just a singular ceiling for color values that a client application should clamp its values to. In this case, the headroom will be set with that maximum value, but the color points and luminance values will be unset, since they are unknown.

The SDR white level is used by SDL's internal renderers, and Windows explicitly provides the value it wants used for this, so it needs to be  passed through.